components/site/src/lib.rs

```diff
@@ -867,12 +863,48 @@ impl Site {
             taxonomies.push(terms);
         }
 
-        context.insert("taxonomies", &taxonomies);
-        context.insert("config", &self.config);
-
-        let sitemap = &render_template("sitemap.xml", &self.tera, context, &self.config.theme)?;
-
-        create_file(&self.output_path.join("sitemap.xml"), sitemap)?;
+        // Count total number of urls to include in sitemap
+        let total_number = pages.len() + sections.len() + taxonomies.len();
+        let sitemap_limit = 30000;
+
+        if total_number < sitemap_limit {
+            // Create one sitemap with all pages, sections and taxonomies
+            let mut context = Context::new();
+            context.insert("pages", &pages);
+            context.insert("sections", &sections);
+            context.insert("taxonomies", &taxonomies);
```

Comment:
you need to have the same context variables here than when rendering a chunked sitemaps, otherwise we cannot use the same template.